### PR TITLE
Fix #7324: Emit a warning if multiple files for same document found

### LINKED ISSUE #7324: Files with same name but different extensions leads to unexpected behaviour
**Describe the bug**

If there are multiple files with the same name but different file extensions;
sphinx will silently choose only one to parse:

**To Reproduce**

Given I have an extension installed to parse  `md`

with:
```
index.rst
a.md
a.rst
```

index.rst:

```restructuredtext
.. toctree::
    a.md
```

This will actually include `a.rst` in the document, not `a.md`

**Expected behavior**

Ideally you would have a config option to specify the order of preference for file extensions,
or if not set, a warning would be logged.

**Your project**
Link to your sphinx project, or attach zipped small project sample.

First noted in:
https://github.com/ExecutableBookProject/MyST-NB/pull/82#issuecomment-599255775

**Screenshots**

**Environment info**
- OS: Mac
- Python version: 3.7.6
- Sphinx version: 2.4.4
- Sphinx extensions:  `myst-parser`

**Additional context**

- [e.g. URL or Ticket]

### PR DESCRIPTION
**Pull request by tk0miya**
### Feature or Bugfix
- Feature
- Bugfix

### Purpose
- refs: #7324

### CODE REVIEW

File: sphinx/project.py [hunk 1]
```diff
@@ -55,7 +56,13 @@ def discover(self, exclude_paths=[]):
         for filename in get_matching_files(self.srcdir, excludes):  # type: ignore
             docname = self.path2doc(filename)
             if docname:
-                if os.access(os.path.join(self.srcdir, filename), os.R_OK):
+                if docname in self.docnames:
+                    pattern = os.path.join(self.srcdir, docname) + '.*'
+                    files = [relpath(f, self.srcdir) for f in glob(pattern)]
+                    logger.warning(__('multiple files found for the document "%s": %r\n'
+                                      'Use %r for the build.'),
+                                   docname, files, self.doc2path(docname), once=True)
```
Comment (tk0miya) [inline sphinx/project.py:64]: Now `project.discover()` is called twice. So I needed to improve `logger.warning()` to suppress this warning on the　second call.

Comment (shimizukawa) [reply sphinx/project.py:64]: Looks good.

**Review by shimizukawa — APPROVED** [review]
LGTM!

**Comment by tk0miya** [comment]
Thank you for reviewing!

**Pull request merged**